src/webgpu/api/validation/state/device_lost/destroy.spec.ts

```diff
@@ -443,126 +497,446 @@ Tests creating query sets on destroyed device.
 
 g.test('command,renderPass,draw')
   .desc(
     `
-Tests encoding and finishing a simple valid render pass on destroyed device fails.
+Tests encoding and finishing a simple valid render pass on destroyed device.
   - Binds valid pipeline and bindgroups, then draws
-  - Tests that finishing encoding fails on destroyed device
-  - Tests that submitting command fails on destroyed device
+  - Tests finishing encoding on destroyed device
+  - Tests submitting command on destroyed device
   `
   )
-  .unimplemented();
+  .params(u =>
+    u.beginSubcases().combine('stage', kCommandValidationStages).combine('awaitLost', [true, false])
+  )
+  .fn(async t => {
+    const { stage, awaitLost } = t.params;
+    const vShader = t.device.createShaderModule({ code: t.getNoOpShaderCode('VERTEX') });
+    const fShader = t.device.createShaderModule({ code: t.getNoOpShaderCode('FRAGMENT') });
+    const pipeline = t.device.createRenderPipeline({
+      vertex: { module: vShader, entryPoint: 'main' },
+      fragment: {
+        module: fShader,
+        entryPoint: 'main',
+        targets: [{ format: 'rgba8unorm', writeMask: 0 }],
+      },
+    });
+    await t.executeCommandsAfterDestroy(stage, awaitLost, 'render pass', maker => {
+      maker.encoder.setPipeline(pipeline);
+      maker.encoder.draw(0);
+      return maker;
+    });
+  });
 
 g.test('command,renderPass,renderBundle')
   .desc(
     `
-Tests encoding and drawing a render pass including a render bundle on destroyed device fails.
+Tests encoding and drawing a render pass including a render bundle on destroyed device.
   - Binds valid pipeline and bindgroups, executes render bundle, then draws
-  - Tests that finishing encoding fails on destroyed device
-  - Tests that submitting command fails on destroyed device
+  - Tests finishing encoding on destroyed device
+  - Tests submitting command on destroyed device
   `
   )
-  .unimplemented();
+  .params(u =>
+    u.beginSubcases().combine('stage', kCommandValidationStages).combine('awaitLost', [true, false])
+  )
+  .fn(async t => {
+    const { stage, awaitLost } = t.params;
+    const vShader = t.device.createShaderModule({ code: t.getNoOpShaderCode('VERTEX') });
+    const fShader = t.device.createShaderModule({ code: t.getNoOpShaderCode('FRAGMENT') });
+    const pipeline = t.device.createRenderPipeline({
+      vertex: { module: vShader, entryPoint: 'main' },
+      fragment: {
+        module: fShader,
+        entryPoint: 'main',
+        targets: [{ format: 'rgba8unorm', writeMask: 0 }],
+      },
+    });
+    await t.executeCommandsAfterDestroy(stage, awaitLost, 'render bundle', maker => {
+      maker.encoder.setPipeline(pipeline);
+      maker.encoder.draw(0);
+      return maker;
+    });
+  });
 
 g.test('queue,writeBuffer')
   .desc(
     `
-Tests writeBuffer on queue on destroyed device fails.
+Tests writeBuffer on queue on destroyed device.
+  `
+  )
+  .params(u =>
+    u
+      .combine('arrayType', kTypedArrayBufferViewKeys)
```

Comment:
now I see why you did the typed array changes.
we can keep them, but TBH not sure if it's worth testing writeBuffer with different types of typed arrays. 
The other writeBuffer test is a validation test, so it makes sense that we need to test with all the data types. For here, the destroy behavior shouldn't depend on the data type. 
Testing several different sizes would still be helpful though in case that puts the memory allocators in different states (I guess different types implicitly results in different sizes)

Reply:
but I think we should still remove the arrayType param since it shouldn't be important for the destroy test.